habitat_baselines/common/base_trainer.py

```diff
@@ -32,15 +41,77 @@ class BaseRLTrainer(BaseTrainer):
     r"""Base trainer class for RL trainers. Future RL-specific
     methods should be hosted here.
     """
+    device: torch.device
+    config: Config
+    video_option: List[str]
 
-    def __init__(self, config):
+    def __init__(self, config: Config):
         super().__init__()
+        assert config is not None, "needs config file to initialize trainer"
         self.config = config
 
     def train(self) -> None:
         raise NotImplementedError
 
     def eval(self) -> None:
+        r"""Main method of trainer evaluation. Calls _eval_checkpoint() that
+        is specified in Trainer class that inherits from BaseRLTrainer
+
+        Returns:
+            None
+        """
+        self.device = torch.device("cuda", self.config.TORCH_GPU_ID)
+
+        if "tensorboard" in self.config.VIDEO_OPTION:
+            assert (
+                len(self.config.TENSORBOARD_DIR) > 0
+            ), "Must specify a tensorboard directory for video display"
+        if "disk" in self.config.VIDEO_OPTION:
+            assert (
+                len(self.config.VIDEO_DIR) > 0
+            ), "Must specify a directory for storing videos on disk"
+
+        with TensorboardWriter(
+            self.config.TENSORBOARD_DIR, purge_step=0, flush_secs=30
```

Comment:
we should make the `purge_step` and `flush_secs` as class attributes instead of constants.